expense-rescript/src/context/GlobalContext.res

```diff
@@ -0,0 +1,59 @@
+@react.component
+type transaction = {id: int, text: string, price: int}
+@val @scope("JSON") external parseIntoMyData: string => array<transaction> = "parse"
+type state = array<transaction>
+let state = []
+type contextType = {
+  transactions: array<transaction>,
+  addTransactionHandler: transaction => unit,
+}
+let context = React.createContext({
+  let saveTransactions = Dom.Storage2.getItem(Dom.Storage2.sessionStorage, "saveTransactions")
+  {
+    transactions: switch saveTransactions {
+    | None => []
+    | Some(value: string) => parseIntoMyData(value)
+    },
+    addTransactionHandler: _ => (),
+  }
+})
```

Comment:
Create this as `defaultValue`
```suggestion
let context = React.createContext(defaultValue)
```

Reply:
Pass the saved transactions wherever initialising the context. For default value pass `transactions` as an empty array.